### PR TITLE
Add support for SDXL LoRA models in diffusers format.

### PR DESCRIPTION
**Pull request by RyanJDick**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ ] No

## Description

This PR introduces 2 related changes:

1. This PR adds support for SDXL LoRA models in the diffusers format. There are 2 popular SDXL LoRA weight formats: Stability AI format and diffusers format (produced by diffusers, kohya_ss, and soon invoke-training). Prior to this change, InvokeAI only supported the Stability AI format, and trying to load a LoRA model in diffusers format would raise an exception.
2. This PR fixes a bug in the current version of InvokeAI that causes all text encoder LoRA weights to be ignored. In other words, prior to this change InvokeAI was only loading UNet LoRA weights for SDXL models.

## Related Tickets & Documents

Relevant Discord conversation where someone encountered this issue: https://discord.com/channels/1020123559063990373/1133929466763161650/1139504370627661896

## QA Instructions, Screenshots, Recordings

I tested the following models:

### Voxel XL
- https://civitai.com/models/118536
- Trigger word: "voxel style"
- Format: Stability AI
- This LoRA was already 'working' before this PR. But the performance seems to be improved after fixing the text encoder bug.
- Before change:
image
- After change (Same seed and configs):
image

### SDXL William Eggleston Style
- https://civitai.com/models/119667
- Trigger words: "by william eggleston"
- Format: diffusers
- Before this change: raised Exception
- After change:
image

### Angelina Jolie SDXL
- https://civitai.com/models/125123/angelina-jolie-sdxl
- Trigger words: "ohwx woman"
- Format: diffusers
- Before this change: raised Exception
- After change:
image

### Embroidery Style - SDXL
- https://civitai.com/models/122308
- Trigger words: None?
- Format: diffusers
- Before this change: raised Exception
- After change: **The LoRA runs, but does not match the results on CivitAI**. I'm still trying to figure out why.
image

## Added/updated tests?

- [ ] Yes
- [x] No : From what I can see, there is 0 test coverage of the LoRA loading code right now. Do we have any plans to change this?

### CODE REVIEW

File: invokeai/backend/model_management/models/lora.py [hunk 1]
```diff
@@ -482,30 +485,61 @@ def calc_size(self) -> int:
...
+        # Get a sorted list of Stability AI UNet keys so that we can efficiently search for keys with matching prefixes.
+        # For example, we want to efficiently find `input_blocks_4_1` in the list when searching for
+        # `input_blocks_4_1_proj_in`.
+        stability_unet_keys = list(SDXL_UNET_STABILITY_TO_DIFFUSERS_MAP)
+        stability_unet_keys.sort()
+
         new_state_dict = dict()
         for full_key, value in state_dict.items():
-            if full_key.startswith("lora_te1_") or full_key.startswith("lora_te2_"):
-                continue  # clip same
```
Comment (RyanJDick) [inline invokeai/backend/model_management/models/lora.py:489]: This is the bug where text_encoder weights were previously being discarded rather than added to `new_state_dict`.

**Comment by RyanJDick** [comment]
If anyone has ideas as to why the Embroidery Style LoRA isn't working as expected, I'd love to hear them. I'm still investigating this, but not sure if it's worth holding up this PR over it.

**Review by StAlKeR7779 — APPROVED** [review]
